natiami/assignment4/FindIslands.cpp

```diff
@@ -0,0 +1,45 @@
+#include"FindIslands.h"
+
+FindIslands::FindIslands(){}
+
+FindIslands::FindIslands(int nr_row, int nr_column, vector<vector<bool> >& mapOfTiles) : nr_row(nr_row), nr_column(nr_column), mapOfTiles(mapOfTiles){
+}
+
+bool FindIslands::checkBounds( int i, int j){
+	return i >= 0 && i < nr_row && j >= 0 && j < nr_column;
+}
+
+void FindIslands::visitOneIsland(vector<vector<bool> >& visited, int i, int j){
+	visited[i][j] = true;
+	vector<pair<int, int> > moves{ { 0, 1 }, { 1, 0 }, { 0, -1 }, { -1, 0 } };
+	int row, column;
+
+	for (size_t k = 0; k < moves.size(); k++)
+	{
+		row = i + moves[k].first;
+		column = j + moves[k].second;
+		if (checkBounds(row, column)){
+			if (!visited[row][column] && mapOfTiles[row][column])
+				visitOneIsland( visited, row, column);
+		}
+	}
+
+}
+
+int FindIslands::findNumberOfIslands(){
+
+	vector<vector<bool> > visited(nr_row, vector<bool>(nr_column));
+	int nr_island = 0;
+
+	for (size_t i = 0; i < nr_row; i++)

```

Comment:
Note that `size_t` is an unsigned int type.
The [Google Stype Guide](https://google.github.io/styleguide/cppguide.html#Integer_Types) recommends avoiding unsigned ints, unless you need the extra values space, even if you are sure that the number is always non-negative.
The main reason is that the compiler implicitly converts signed to unsigned ints and vice versa, overflowing the values, which can cause errors that are easy to miss. For example:
```C++
size_t x = -1;
bool b = x > 0;  // Evaluates to true!!
cout << x; // Prints: 18446744073709551615
```

More info: http://wesmckinney.com/blog/avoid-unsigned-integers/

In this case, you even compare it with `nr_row` which is an int, so it is best to just keep using ints.